sqlglot/dataframe/sql/column.py

```diff
@@ -0,0 +1,330 @@
+        return Column(new_expression)
+
+    def cast(self, dataType: t.Union[str, "DataType"]):
+        """
+        Functionality Difference: PySpark cast accepts a datatype instance of the datatype class
+        Sqlglot doesn't currently replicate this class so it only accepts a string
+        """
+        if isinstance(dataType, DataType):
+            dataType = dataType.simpleString()
+        new_expression = exp.Cast(this=self.column_expression, to=dataType)
+        return Column(new_expression)
+
+    def startswith(self, value: t.Union[str, Column]) -> Column:
+        value = self._lit(value) if not isinstance(value, Column) else value
+        return self.invoke_anonymous_function(self, "STARTSWITH", value)
+
+    def endswith(self, value: t.Union[str, Column]) -> Column:
+        value = self._lit(value) if not isinstance(value, Column) else value
+        return self.invoke_anonymous_function(self, "ENDSWITH", value)
+
+    def rlike(self, regexp: str) -> Column:
+        return self.invoke_expression_over_column(self, exp.RegexpLike, expression=self._lit(regexp).expression)
+
+    def like(self, other: str):
+        return self.invoke_expression_over_column(self, exp.Like, expression=self._lit(other).expression)
+
+    def ilike(self, other: str):
+        return self.invoke_expression_over_column(self, exp.ILike, expression=self._lit(other).expression)
+
+    def substr(self, startPos: t.Union[int, Column], length: t.Union[int, Column]) -> Column:
+        startPos = self._lit(startPos) if not isinstance(startPos, Column) else startPos
+        length = self._lit(length) if not isinstance(length, Column) else length
+        return Column.invoke_expression_over_column(
+            self, exp.Substring, start=startPos.expression, length=length.expression
+        )
+
+    def isin(self, *cols: t.Union["Primitives", t.Iterable["Primitives"]]):
+        cols = flatten(cols) if isinstance(cols[0], (list, set, tuple)) else cols
+        expressions = [self._lit(x).expression for x in cols]
+        return Column.invoke_expression_over_column(self, exp.In, expressions=expressions)
+
+    def between(
+        self,
+        lowerBound: t.Union[Column, "Literals", "DateTimeLiteral", "DecimalLiteral"],
+        upperBound: t.Union[Column, "Literals", "DateTimeLiteral", "DecimalLiteral"],
+    ) -> Column:
+        lower_bound_exp = self._lit(lowerBound) if not isinstance(lowerBound, Column) else lowerBound
+        upper_bound_exp = self._lit(upperBound) if not isinstance(upperBound, Column) else upperBound
+        return Column(
+            exp.Between(this=self.column_expression, low=lower_bound_exp.expression, high=upper_bound_exp.expression)
+        )
+
+    def over(self, window: WindowSpec) -> Column:
+        window_args = window.expression.args
+        window_args["this"] = self.column_expression
+        return Column(exp.Window(**window_args))
```

Comment:
can you add all places where you instantiate an expression to have kwargs, like this=